src/components/Form/CreateProfileForm/Documents/FileRow.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useRef, useState } from "react";
+import { Button, Icon, Progress } from "design-react-kit";
+import { saveAs } from "file-saver";
+import { tryCatch } from "fp-ts/lib/TaskEither";
+import { toError } from "fp-ts/lib/Either";
+import { useTooltip, Severity } from "../../../../context/tooltip";
+import DocumentIcon from "../../../../assets/icons/document.svg";
+import Api from "../../../../api";
+import DocumentSuccess from "../../../../assets/icons/document-success.svg";
+import { Document } from "../../../../api/generated";
+import DeleteDocument from "./DeleteDocument";
+
+type Props = {
+  type: string;
+  label: string;
+  uploadedDoc?: Document;
+  getFiles: () => void;
+  agreementId: string;
+};
+
+const FileRow = ({
+  type,
+  label,
+  uploadedDoc,
+  getFiles,
+  agreementId
+}: Props) => {
+  const refFile = useRef<any>();
+  const [loadingTemplate, setLoadingTemplate] = useState(false);
+  const [loadingDoc, setLoadingDoc] = useState(false);
+  const { triggerTooltip } = useTooltip();
+
+  const handleClick = () => {
+    refFile.current.click();
+  };
+
+  const getTemplates = async () => {
+    setLoadingTemplate(true);
+    await tryCatch(
+      () =>
+        Api.DocumentTemplate.downloadDocumentTemplate(agreementId, type, {
+          headers: {
+            "Content-Type": "application/pdf"
+          },
+          responseType: "arraybuffer"
+        }),
+      toError
+    )
+      .map(response => response.data)
+      .fold(
+        () => setLoadingTemplate(false),
+        response => {
+          if (response) {
+            const blob = new Blob([response], { type: "application/pdf" });
+            saveAs(blob, label);
+          }
+          setLoadingTemplate(false);
+        }
+      )
+      .run();
+  };
+
+  const addFile = async (files: any) => {
+    setLoadingDoc(true);
+    await tryCatch(
+      () => Api.Document.uploadDocument(agreementId, type, files[0]),
+      toError
+    )
+      .fold(
+        () => {
+          setLoadingDoc(false);
+        },
+        response => {
+          if (response.status) {
+            getFiles();
+          } else {
+            triggerTooltip({
+              severity: Severity.DANGER,
+              text: "Caricamento del file fallito"
+            });
```

Comment:
I'd rather prefer ternary operator instead of `if block`

Reply:
Fixed.